eslint.config.mjs

```diff
@@ -0,0 +1,172 @@
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+      globals: {
+        ...globals.browser,
+        ...globals.node,
+        es6: true,
+      },
+    },
+    plugins: {
+      '@next/next': pluginNext,
+      '@typescript-eslint': typescriptEslint,
+      '@stylistic': stylistic,
+      'react-hooks': reactHooks,
+      import: importPlugin,
+      prettier,
+      jsdoc,
+    },
+    rules: {
+      ...pluginNext.configs.recommended.rules,
+      ...pluginNext.configs['core-web-vitals'].rules,
+      '@next/next/no-html-link-for-pages': 'off',
+      '@next/next/no-img-element': 'off',
+      '@next/next/no-sync-scripts': 'off',
+      '@next/next/no-assign-module-variable': 'off',
+      // Include ESLint recommended rules
+      ...js.configs.recommended.rules,
+      // Include JSDoc recommended rules
+      ...jsdoc.configs.recommended.rules,
+      // JSDoc relaxations
+      'jsdoc/newline-after-description': 'off',
+      'jsdoc/require-property-description': 'off',
+      'jsdoc/require-param-description': 'off',
+      'jsdoc/require-returns': 'off',
+      'jsdoc/no-undefined-types': 'off',
+      'jsdoc/require-returns-type': 'off',
+
+      // General
+      'no-use-before-define': 'off',
+      'no-useless-escape': 'off',
+      'no-unused-vars': ['error'],
+      'no-unused-expressions': [
+        'error',
+        {
+          allowShortCircuit: true,
+          allowTernary: true,
+        },
+      ],
+      'spaced-comment': 'error',
+      curly: ['error', 'multi-line'],
+      'eol-last': ['error', 'always'],
+      'guard-for-in': 'error',
+      'no-unused-labels': 'error',
+      'no-caller': 'error',
+      'no-bitwise': 'error',
+      'no-multiple-empty-lines': 'error',
+      'no-new-wrappers': 'error',
+      'no-eval': 'error',
+      'dot-notation': 'error',
+      'no-trailing-spaces': 'error',
+      'brace-style': 'error',
+      quotes: ['error', 'single'],
+      radix: 'error',
+      'default-case': 'error',
+      eqeqeq: 'error',
+      'jsx-quotes': ['error', 'prefer-double'],
+
+      // Formatting
+      'prettier/prettier': 'off', // handled via specific rules below
+      '@stylistic/space-before-function-paren': [
+        'error',
+        {
+          anonymous: 'never',
+          named: 'never',
+          asyncArrow: 'always',
+        },
+      ],
+      'react-hooks/rules-of-hooks': 'error',
+      'react-hooks/exhaustive-deps': 'warn',
+      'import/no-anonymous-default-export': 'error',
+
+      // Prettier conflict resolution - must come LAST to override conflicting rules
+      ...eslintConfigPrettier.rules,
+    },
+  },
+
+  // TypeScript-specific overrides
+  {
+    files: ['packages/**/*.{ts,tsx}'],
+    rules: {
+      '@typescript-eslint/naming-convention': [
+        'error',
+        {
+          selector: 'typeLike',
+          format: ['PascalCase'],
+          custom: { regex: '^I[A-Z]', match: false },
+        },
+      ],
+      'no-undef': 'off',
```

Comment:
Since TypeScript has its own type checking, this rule might not be necessary. But its upto you to remove.